activesupport/test/claims_test.rb

```diff
@@ -0,0 +1,154 @@
+require 'abstract_unit'
+require 'active_support/time'
+
+class ClaimsTest < ActiveSupport::TestCase
+  def setup
+    @value = 'payload'
+    @claims = ActiveSupport::Claims.new(payload: @value, for: 'test', expires_at: Time.utc(2022))
+  end
+
+  def test_to_hash
+    hash = { pld: 'payload', for: 'test', exp: '2022-01-01T00:00:00.000Z' }
+    assert_equal hash, @claims.to_h
+  end
+
+  def test_verify_returns_value_with_valid_claims
+    assert_equal @value, ActiveSupport::Claims.verify!(@claims.to_h, for: 'test')
+  end
+
+  def test_verify_returns_value_with_default_claims
+    claims = ActiveSupport::Claims.new(payload: @value)
+    assert_equal @value, ActiveSupport::Claims.verify!(claims.to_h)
+  end
+end
+
+class ClaimsPurposeTest < ActiveSupport::TestCase
+  def setup
+    @value = 'payload'
+    @claims = ActiveSupport::Claims.new(payload: @value, for: 'test')
+  end
+
+  def test_default_purpose_without_for
+    assert_equal 'universal', ActiveSupport::Claims.new(payload: @value).purpose
+  end
+
+  def test_verify_exception_on_invalid_purpose
+    assert_raise(ActiveSupport::Claims::InvalidClaims) do
+      ActiveSupport::Claims.verify!(@claims.to_h, for: 'different_purpose')
+    end
+  end
+
+  def test_equal_only_with_same_purpose
+    assert_equal @claims, ActiveSupport::Claims.new(payload: @value, for: 'test')
+    assert_not_equal @claims, ActiveSupport::Claims.new(payload: @value, for: 'login')
+    assert_not_equal @claims, ActiveSupport::Claims.new(payload: @value)
+  end
+end
+
+class ClaimsExpirationTest < ActiveSupport::TestCase
+  def setup
+    @value = 'payload'
```

Comment:
I feel like you're flip flopping between value and payload. Let's stick to payload as much as we can :smile: